rsocket-core/src/main/java/io/rsocket/frame/RequestFlyweight.java

```diff
@@ -31,7 +32,7 @@ ByteBuf encode(
       ByteBuf data) {
     int flags = 0;
 
-    if (metadata != null) {
+    if (metadata != null && Unpooled.EMPTY_BUFFER != metadata) {
```

Comment:
Isn't EMPTY_BUFFER the "" case? and null is the case where flag should be unset?

Reply:
yeah I  that is correct - that would explain the other thing I changed in the unit - it was kind of early - I will go through and check the flyweights.